src/python/pants/jvm/target_types.py

```diff
@@ -23,45 +23,51 @@ class JvmArtifactGroupField(StringField):
     alias = "group"
     required = True
     help = (
-        "The 'group' part of a Maven-compatible coordinate to a third-party jar artifact. For the jar coordinate "
-        "com.google.guava:guava:30.1.1-jre, the group is 'com.google.guava'."
+        "The 'group' part of a Maven-compatible coordinate to a third-party JAR artifact.\n\n"
+        "For the JAR coordinate `com.google.guava:guava:30.1.1-jre`, the group is "
+        "`com.google.guava`."
     )
 
 
 class JvmArtifactArtifactField(StringField):
     alias = "artifact"
     required = True
     help = (
-        "The 'artifact' part of a Maven-compatible coordinate to a third-party jar artifact. For the jar coordinate "
-        "com.google.guava:guava:30.1.1-jre, the artifact is 'guava'."
+        "The 'artifact' part of a Maven-compatible coordinate to a third-party JAR artifact.\n\n"
+        "For the JAR coordinate `com.google.guava:guava:30.1.1-jre`, the artifact is `guava`."
     )
 
 
 class JvmArtifactVersionField(StringField):
     alias = "version"
     required = True
     help = (
-        "The 'version' part of a Maven-compatible coordinate to a third-party jar artifact. For the jar coordinate "
-        "com.google.guava:guava:30.1.1-jre, the version is '30.1.1-jre'."
+        "The 'version' part of a Maven-compatible coordinate to a third-party JAR artifact.\n\n"
+        "For the JAR coordinate `com.google.guava:guava:30.1.1-jre`, the version is `30.1.1-jre`."
     )
 
 
 class JvmArtifactUrlField(StringField):
     alias = "url"
     required = False
     help = (
-        "A URL that points to the location of this artifact. If specified, Pants will not fetch this artifact "
-        "from default maven repositories, and instead fetch the artifact from this URL. To use default maven "
-        "repositories, do not set this value. \n\nNote that `file:` URLs are not supported due to Pants' "
-        "sandboxing feature. To use a local `JAR` file, use the `jar` field instead."
+        "A URL that points to the location of this artifact.\n\n"
+        "If specified, Pants will not fetch this artifact from default Maven repositories, and "
+        "will instead fetch the artifact from this URL. To use default maven "
+        "repositories, do not set this value.\n\n"
+        "Note that `file:` URLs are not supported. Instead, use the `jar` field."
```

Comment:
I removed the part about "why" `file:` URLs don't work. Benjy and Stu have pointed out to me in the past it's sometimes information overload to explain why, and I suspect that's the case here.